### PR TITLE
Set fdelay first to allow growing maxage

### PR DESCRIPTION
**Pull request by lagerspetz-lempea**
Fixes https://github.com/mstpd/mstpd/issues/86

### CODE REVIEW

**Comment by commodo** [comment]
Thanks for the patch.

Will merge this in a couple of days if no objections.

File: utils/ifupdown.sh.in [hunk 1]
```diff
@@ -107,14 +107,14 @@ if [ "$MODE" = 'start' ]; then
...
 
-  if [ "$IF_MSTPCTL_MAXAGE" ]; then
-    "$mstpctl" setmaxage "$IFACE" "$IF_MSTPCTL_MAXAGE"
-  fi
-
   if [ "$IF_MSTPCTL_FDELAY" ]; then
     "$mstpctl" setfdelay "$IFACE" "$IF_MSTPCTL_FDELAY"
   fi
 
+  if [ "$IF_MSTPCTL_MAXAGE" ]; then
```
Comment (commodo) [inline utils/ifupdown.sh.in:114]: one note it just occurred to me;
maybe it's worth adding a comment here describing the issue this solves;
maybe also in the commit;

over time, we tend to forget things; i usually search for things better in git history via CLI;

Comment (lagerspetz-lempea) [reply utils/ifupdown.sh.in:114]: How about what I mentioned in the bug report, what if someone now wants to decrease fdelay below 10 in the interface definition, and that fails? That would require setting fdelay again after maxage. What if we set fdelay twice: once before and once after maxage?

Comment (mg964) [reply utils/ifupdown.sh.in:114]: Did you see my comment over on the issues thread?

https://github.com/mstpd/mstpd/issues/86#issuecomment-633889891

My shell scripting skills are limited, don't know how complicated that would be.

Comment (lagerspetz-lempea) [reply utils/ifupdown.sh.in:114]: Thanks, didn't see it. I think it's doable.

Comment (lagerspetz-lempea) [reply utils/ifupdown.sh.in:115]: I have yet to test it in my environment, but please check it when you have time.

Comment (commodo) [reply utils/ifupdown.sh.in:115]: yes, this should run on a standard bourne shell (i.e. /bin/sh), but i'm a bit unclear on what that shell supports;
even busybox's shell (ash) seems to be able to handle $()
in this shell, there are already uses of $(), so you could use them;
if we get complaints to go for an older form, we might update to something like that;

regarding, `let`, i did run on some issues with it not working on bourne; so, using `expr` might be a good idea;

Comment (lagerspetz-lempea) [reply utils/ifupdown.sh.in:115]: I found that $(()) works (with help of shellcheck). This `should` now pass CI.

**Comment by commodo** [comment]
can you squash all commits (except for `Add ignores generated by CI build`) into a single commit?
since some of them are fix-commits, they usually don't matter in master, if we have committed them into master [yet]

**Comment by lagerspetz-lempea** [comment]
Sure, I will squash. Testing a bit first.

**Comment by lagerspetz-lempea** [comment]
Works for me --- merge when ready.

**Comment by commodo** [comment]
will let this sit for a couple more days;
if no objections, will merge;

**Comment by commodo** [comment]
many thanks for this btw :)

File: utils/ifupdown.sh.in [hunk 2]
```diff
@@ -107,13 +107,29 @@ if [ "$MODE" = 'start' ]; then
...
+  age=0
   if [ "$IF_MSTPCTL_MAXAGE" ]; then
-    "$mstpctl" setmaxage "$IFACE" "$IF_MSTPCTL_MAXAGE"
+    age="$IF_MSTPCTL_MAXAGE"
   fi
 
-  if [ "$IF_MSTPCTL_FDELAY" ]; then
-    "$mstpctl" setfdelay "$IFACE" "$IF_MSTPCTL_FDELAY"
-  fi
+  curfwddly=$( "$mstpctl" showbridge "$IFACE" forward-delay )
```
Comment (mg964) [inline utils/ifupdown.sh.in:119]: Shouldn't this be "bridge-forward-delay", i.e. the local value not the value from the root bridge?

Looks like I used the wrong value in the original issue post.

Comment (lagerspetz-lempea) [reply utils/ifupdown.sh.in:119]: I was not sure which one refers to the local value. I will change this.

**Comment by commodo** [comment]
Thanks again :)

**Comment by lagerspetz-lempea** [comment]
Thanks, good to get this in. What is your release policy, is it possible to increase the deb version and make the current master 0.0.9? Or do you have something larger in mind for that @commodo ?

**Comment by commodo** [comment]
Well, we don't have a clear release policy, as things don't change too much.
This package is useful only to a handful of users (no idea how many).
I think we can make a new release, since it's been a while since the last.

If you have anything to add to the new one, it's a good time to add it now, and we can make a new one.

**Comment by commodo** [comment]
well, good news is: 0.0.9 will be the first version to have an official debian package :)
no idea how fast we'll get this into debian sid;
but let's see

**Pull request merged**